meson.build

```diff
@@ -75,44 +75,62 @@ configinc = include_directories('.')
 cc = meson.get_compiler('c')
 dl_dep = cc.find_library('dl', required : false)
 
-WITH_DRM = not get_option('disable_drm') and (host_machine.system() != 'windows')
-libdrm_dep = dependency('libdrm', version : '>= 2.4.60', required : (host_machine.system() != 'windows'))
+backends = ''
+
+WITH_DRM = false
+if not get_option('with_drm').disabled()
+  libdrm_dep = dependency('libdrm', version : '>= 2.4.60', required : (host_machine.system() != 'windows'))
```

Comment:
If with_drm is set to auto, the libdrm dependency should be optional. Here you make it required on non-windows.

I would suggest doing instead:
```
libdrm_dep = dependency('libdrm', version : '>= 2.4.60', required : get_option('with_drm'))
WITH_DRM = libdrm_dep.found()
if WITH_DRM
  backends += ' drm'
endif
```

I don't the it matters to check the platform, if someone build with -Dwith_drm=enabled on Windows, they'll get an error because libdrm is not found and that's fine IMHO.

Reply:
> If with_drm is set to auto, the libdrm dependency should be optional. 

Not quite: libdrm is a hard requirement for all non-win32 backends - android, x11 (dri3) and by extension glx, wayland (wl_drm) or drm itself - they all use the `va_drm_utils.c` helper which needs libdrm.

>I don't the it matters to check the platform, if someone build with -Dwith_drm=enabled on Windows, they'll get an error because libdrm is not found and that's fine IMHO.

There is a word missing above, it seems. Regardless - as long as the author @sivileri is fine it works for me.

Reply:
For Windows it should still be fine, as long as nobody does `-Dwith_drm` without `libdrm`. I don't have any objections with making `libdrm` optional, as long as we are aware, as @evelikov points out, that doing the proposed change will make `libdrm` optional in all non-windows environments with have a hard dependency to it.

Reply:
Sounds like we're fine as-is them.

Reply:
@evelikov I'm not libva expert, but it still feels wrong. With this PR, on linux, you can do `-Dwith_drm=disabled` and it will not use libdrm at all, so that dep is optional even on non-Windows platforms? If it makes sense to have libdrm disabled for a linux build, then `with_drm=auto` should allow libdrm dependency to be not-found. Only `-Dwith_drm=enabled` should force having libdrm on Linux IMHO. Otherwise a tristate enabled/disabled/auto does not make much sense, in your current implementation auto has the same effect as enabled. Unless there is something I missed?

Reply:
Possibly with_drm should not be an option at all and we can remove the ` if not get_option('with_drm').disabled()` condition, to not allow disabling libdrm?

Reply:
Your wording threw me off there for a moment but you're spot on.

In particular `-D with_drm=disabled` is borked - the backend is correctly not selected for building, yet building the x11, wayland... ones fail, since meson sees the libdrm dependency as unresolved.

Will update in a moment to drop the explicit drm toggle - it makes zero sense to have it. Thanks for the suggestion and patience.